### PR TITLE
feat: write limit overlays

### PR DESCRIPTION
**Pull request by ShmuelLotman**
Closes #10955

This introduces more detailed and functional write limit overlays that provide an opportunity for users to upgrade and/or submit requests for increases if they are PAYG. This is a major improvement over our previous implementation, which did.... nothing. 
Screen Shot 2021-06-28 at 10 40 56 AM
Screen Shot 2021-06-28 at 10 42 46 AM
Screen Shot 2021-06-28 at 10 39 24 AM

### CODE REVIEW

File: src/cloud/components/WriteLimitOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  const {onClose} = useContext(OverlayContext)
+
+  const {id} = useSelector(getOrg)
+
+  const handleSubmit = (): void => {
+    event('limit.requestincrease.writes', {reason: limitReason})
+
+    window.open(
+      `mailto:support@influxdata.com?subject=Request%20Query%20Write%20Limit%20Increase&body=Organization ID%3A%20${id}%0D%0A%0D%0AUse-Case Details:%0D%0A${limitReason}`
```
Comment (Asalem1) [inline src/cloud/components/WriteLimitOverlay.tsx:42]: just checking, the whitespace between Organization & ID are handled ok right? Same for Case Details

File: src/overlays/components/OverlayController.tsx [hunk 2]
```diff
@@ -150,7 +154,7 @@ const OverlayProvider: FunctionComponent = props => {
 
   const closer = useCallback(() => {
     dispatch(dismissOverlay())
-    if (onClose) {
+    if (onClose && typeof onClose === 'function') {
       onClose()
```
Comment (Asalem1) [inline src/overlays/components/OverlayController.tsx:158]: can onClose be anything else?

**Review by Asalem1 — APPROVED** [review]

File: src/cloud/components/WriteLimitOverlay.tsx (hunk 1 above)
Comment (Asalem1) [reply src/cloud/components/WriteLimitOverlay.tsx:42]: Ignore this, based on the pics it looks like it's kosher

File: src/overlays/components/OverlayController.tsx (hunk 2 above)
Comment (ShmuelLotman) [reply src/overlays/components/OverlayController.tsx:158]: yes it can

Comment (Asalem1) [reply src/overlays/components/OverlayController.tsx:158]: Wait, why? How?

Comment (ShmuelLotman) [reply src/overlays/components/OverlayController.tsx:158]: still trying to figure that out, but somehow onClose becomes an action object. I figured that ill look into it, but in the meantime, if we are going to have a conditional that allows us to call a function, we should make sure its a function regardless

Comment (ShmuelLotman) [reply src/overlays/components/OverlayController.tsx:158]: ah found the issue, pushing up the fix now

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=master)

**Pull request merged**